src/v0/destinations/amazon_audience/utils.js

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+const sha256 = require('sha256');
+const lodash = require('lodash');
+const { ConfigurationError, OAuthSecretError } = require('@rudderstack/integrations-lib');
+const {
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  getSuccessRespEvents,
+  removeUndefinedAndNullAndEmptyValues,
+} = require('../../util');
+
+// Docs: https://developer.x.com/en/docs/x-ads-api/audiences/api-reference/custom-audience-user
+const buildResponseWithUsers = (users, action, config, jobIdList, secret) => {
+  const { audienceId } = config;
+  if (!audienceId) {
+    throw new ConfigurationError('[AMAZON AUDIENCE]: Audience Id not found');
+  }
+  if (!secret?.accessToken) {
+    throw new OAuthSecretError('OAuth - access token not found');
+  }
+  if (!secret?.clientId) {
+    throw new OAuthSecretError('OAuth - Client Id not found');
+  }
+  const externalId = `Rudderstack_${sha256(`${jobIdList}`)}`;
+  const response = defaultRequestConfig();
+  response.endpoint = '';
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.headers = {
+    'Amazon-Advertising-API-ClientId': `${secret.clientId}`,
+    'Content-Type': 'application/json',
+    Authorization: `Bearer ${secret.accessToken}`,
+  };
+  response.body.JSON = {
+    createUsers: {
+      records: [
+        {
+          hashedRecords: users,
+          externalId,
+        },
+      ],
+    },
+    associateUsers: {
+      patches: [
+        {
+          op: action,
+          path: `/EXTERNAL_USER_ID-${externalId}/audiences`,
+          value: [audienceId],
+        },
+      ],
+    },
+  };
+  return response;
+};
+
+/**
+ * This function groups the response list based upon `operation`
+ * @param {*} respList
+ * @returns object
+ */
+const groupResponsesUsingOperation = (respList) => {
+  const eventGroups = lodash.groupBy(respList, (item) => item.message.action);
```

Comment:
Deletes should be before creates

Reply:
handled here
https://github.com/rudderlabs/rudder-transformer/blob/da7d2f8abe6f583f196438b951a4561840a14ea5/src/v0/destinations/amazon_audience/utils.js#L80